clusterversion, *: bump MinSupported from v24.3 to v25.1

michae2 opened the pull request:
Bump MinSupported from v24.3 to v25.1. See individual commits for details.

Epic: REL-2456

Release note: None

michae2 committed:
clusterversion, *: prefix version keys below 25.1 with TODO_Delete_

Epic: REL-2456

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/147130)
<!-- Reviewable:end -->

michae2 committed:
crosscluster/physical: delete subtest of TestAlterReplicationJobErrors

The subtest `alter_replication_dest_priv_24.3` of
`TestAlterReplicationJobErrors` creates a 24.3 cluster. This will no
longer be possible after `clusterversion.MinSupported` is increased to
v25.1.

Epic: REL-2456

Release note: None

michae2 committed:
logictest: remove mixed-24.3 tests

Epic: REL-2456

Release note: None

michae2 committed:
schemachanger/scplan/internal/rules: remove release_24_3

Epic: REL-2456

Release note: None

michae2 committed:
clusterversion: update M4 section of README a little

Epic: REL-2456

Release note: None

michae2 commented:
This could definitely use a second pair of eyes. Here are things I'm not sure about:

- I deleted TestAlterReplicationJobErrors/alter_replication_dest_priv_24.3, but actually I think I want to skip this instead and file an issue?
- I couldn't find the pkg/sql/catalog/bootstrap data for the one table introduced in 24.3 (TableMetadata)
- Not sure I updated storage.pebbleFormatVersionMap correctly. There wasn't a 25.1 mapping so I copied over the 24.3 mapping and left everything else.
- I changed 24.3 to 25.1 in build/teamcity/cockroach/nightlies/sqllogic_corpus_nightly_impl.sh , not sure I was actually supposed to do this.

michae2 committed:
catalog/bootstrap: remove 24.3 data

Epic: REL-2456

Release note: None

michae2 committed:
clusterversion, storage: bump MinSupported from v24.3 to v25.1

Bump MinSupported from v24.3 to v25.1 and update
storage.pebbleFormatVersionMap.

Epic: REL-2456

Release note: None

RaduBerinde reviewed (approved):
Looks great, thanks!

michae2 commented:
Thank you!

bors r=RaduBerinde
